src/Serialization/upgrades/0.11.3.jl

```diff
@@ -0,0 +1,148 @@
+                    for v in dict_value
+                        if v isa String
+                            push!(new_value, v)
+                        else
+                            push!(new_value, upgrade_0_11_3(s, v))
+                        end
+                    end
+                    upgraded_dict[key] = new_value
+                end
+            end
+            return upgraded_dict
+        end
+
+        if dict[:type] == string(backref_sym)
+            backrefed_object = s.objs[UUID(dict[:id])]
+
+            # if the backref points to a string, just use that string
+            # instead of the backref
+            backrefed_object isa String && return backrefed_object
+
+            return dict
+        end
+
+        # this handles types like FlintRationalField that have no associated data
+        if !haskey(dict, :data)
+            # adds object to instance in case it is backrefed
+            if haskey(dict, :id)
+                s.objs[UUID(dict[:id])] = dict
+            end
+
+            return dict
+        end
+
+        # apply upgrade to vector entries
+        # we only do this here since upgrade_0_11_13 is only defined for dicts
+        # we could make this its own function outside of this script
+        if dict[:type] == "Vector"
+            upgraded_vector = []
+            entry_type = nothing
+
+            for entry in dict[:data][:vector]
+                result = upgrade_0_11_3(s, entry)
+
+                # store values in state that are vector entries that aren't backrefs
+                if entry[:type] != string(backref_sym)
+                    if haskey(entry, :id)
+                        s.objs[UUID(entry[:id])] = result
+                    end
+                    @assert entry_type === nothing || entry_type == entry[:type]
+                    entry_type = entry[:type]
+                end
+
+                push!(upgraded_vector, result)
+            end
+
+            # if the objects in the vector are not basic, they are represented
+            # here by dicts, and we do nothing else; if the objects are basic,
+            # they are represented by strings, and we'll add the `entry_type`
+            # to the vector data below...
+            if upgraded_vector[1] isa Dict
+                upgraded_dict = Dict(
+                    :type => "Vector",
+                    :id => dict[:id],
+                    :data => Dict(
+                        :vector => upgraded_vector
+                    )
+                )
+                # add to state
+                s.objs[UUID(dict[:id])] = upgraded_dict
+                return upgraded_dict
+            end
+
+            upgraded_dict = Dict(
+                :type => "Vector",
+                :id => dict[:id],
+                :data => Dict(
+                    :vector => upgraded_vector,
+                    :entry_type => entry_type
+                )
+            )
+            s.objs[UUID(dict[:id])] = upgraded_dict
```

Comment:
We can reduce code duplication a bit:
```suggestion
            upgraded_dict = Dict(
                :type => "Vector",
                :id => dict[:id],
                :data => Dict(
                    :vector => upgraded_vector,
                    :entry_type => entry_type
                )
            )

            # if the objects in the vector are not basic, they are represented
            # here by dicts, and we do nothing else; if the objects are basic,
            # they are represented by strings, and we'll add the `entry_type`
            # to the vector data below...
            if !(upgraded_vector[1] isa Dict)
                upgraded_dict[:data][:entry_type] = entry_type
            end

            # add to state
            s.objs[UUID(dict[:id])] = upgraded_dict
```